src/Mvc/Mvc.RazorPages/src/Infrastructure/PageLoaderMatcherPolicy.cs

```diff
@@ -76,6 +76,12 @@ public Task ApplyAsync(HttpContext httpContext, CandidateSet candidates)
                 var page = endpoint.Metadata.GetMetadata<PageActionDescriptor>();
                 if (page != null)
                 {
+                    if (page.CompiledPageDescriptor != null)
```

Comment:
Can you verify this works correctly with runtime compilation? We use the instance on `CompiledPageAD` to execute the page. Previously `LoadAsync` manages invalidating the compiled AD if the file was modified and it's not clear that happens now

Reply:
I have to ask, do we have any tests for that? Or does it need to be tested manually?

> Previously LoadAsync manages invalidating the compiled AD if the file was modified and it's not clear that happens now

AFAIK, runtime compilation uses https://github.com/dotnet/aspnetcore/blob/5c2cb8b2d4e3864bddb4343c06450d09ba2ffb98/src/Mvc/Mvc.Razor.RuntimeCompilation/src/PageActionDescriptorChangeProvider.cs#L15 to detect changes which will end up making new page descriptors.

Reply:
Spoke about this offline. I misremembered how this currently works, so we do not have to worry about cache invalidation here. I think we might have some integration tests for this, although I'm not 100% certain how well they exercise the edit - > update code paths. A manual verification should help.

Reply:
Tested and it works!